crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -345,24 +345,18 @@ where
         // the RRSIG used to validate a record inherits the outcome of the validation
         // for RRSIGs, we need to use their TYPE_COVERED field instead of `RecordType::RRSIG` as the
         // `RecordType` key in `rrset_proofs`
-        //
-        // The adjusted TTL is determined by RRSIG + RR record it covers.
-        let (record_type, new_ttl) = if let RData::DNSSEC(DNSSECRData::RRSIG(rrsig)) = record.data()
-        {
-            (
-                rrsig.type_covered(),
-                rrsig.authenticated_ttl(record, current_time),
-            )
+        let record_type = if let RData::DNSSEC(DNSSECRData::RRSIG(rrsig)) = record.data() {
+            let record_type = rrsig.type_covered();
+            let new_ttl = rrsig.authenticated_ttl(record, current_time);
+            record.set_ttl(new_ttl);
```

Comment:
I think that with this change the TTL will also get updated when `proof` is not `Secure`?

Reply:
The code has been removed here, `ttl` is only updated when `proof` is `Secure`.